cfme/tests/services/test_service_rbac.py

```diff
@@ -0,0 +1,143 @@
+                             role=role, tenant="My Company")
+
+
+def new_user(appliance, group):
+    collection = appliance.collections.users
+    return collection.create(name='user_{}'.format(fauxfactory.gen_alphanumeric()),
+                             credential=new_credential(),
+                             email='xyz@redhat.com',
+                             groups=group,
+                             cost_center='Workload',
+                             value_assign='Database')
+
+
+@pytest.yield_fixture(scope="module")
+def role_user_group(appliance):
+    role = new_role(
+        appliance=appliance,
+        product_features=[(['Everything'], False)] + [(['Everything', k], True)
+                          for k in ['Access Rules for all Virtual Machines']])
+    group = new_group(appliance=appliance, role=role.name)
+    user = new_user(appliance=appliance, group=group)
+    yield (role, user)
+
+
+@pytest.yield_fixture(scope='function')
+def _myservice(appliance, catalog_item):
+    vm_name = catalog_item.provisioning_data["catalog"]["vm_name"] + '_0001'
+    catalog_item.create()
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    service_catalogs.order()
+    yield catalog_item.name, vm_name
+
+
+def test_service_rbac_no_permission(appliance, role_user_group):
+    """ Test service rbac without user permission """
+    role, user = role_user_group
+    error_message = "The user's role is not authorized for any access, " \
+                    "please contact the administrator!"
+    with error.expected(error_message):
+        appliance.server.login(user)
+
+
+CATALOG_MODULES = ['Catalogs', 'Orchestration Templates', 'Catalog Items', 'Service Catalogs']
+
+
+@pytest.mark.parametrize('permission', CATALOG_MODULES)
+def test_service_rbac_catalog_explorer(appliance, role_user_group, catalog,
+                                       catalog_item, permission):
+    """ Test service rbac with only catalog explorer module permissions"""
+    role, user = role_user_group
+    appliance.server.login_admin()
+    if permission == CATALOG_MODULES[3]:
+        role.update({
+            'product_features': [(['Everything'], True)] + [(['Everything'], False)] +
+                                [(['Everything', 'Services', 'Catalogs Explorer', k], True)
+                                 for k in ['Catalog Items', 'Service Catalogs', 'Catalogs']] +
+                                [(['Everything', 'Services', 'Requests', ], True)] +
+                                [(['Everything', 'Automation', 'Automate', 'Customization'], True)]
+        })
+    else:
+        role.update({
+            'product_features': [(['Everything'], True)] + [(['Everything'], False)] +
+                                [(['Everything', 'Services', 'Catalogs Explorer', k], True)
+                                 for k in [permission]]
+        })
+
+    with user:
+        appliance.server.login(user)
```

Comment:
this is not necessary with user: will do the job

Reply:
@Yadnyawalkya you can remove appliance.server.login(user) if you are using with user: